scipy/optimize/cython_optimize/Newton/newton.c

```diff
@@ -0,0 +1,44 @@
+#include <math.h>
+#include "newton.h"
+#include "../../Zeros/zeros.h"
+
+double
+newton(callback_type func, double p0, callback_type fprime, default_parameters *params, double tol, int maxiter)
```

Comment:
I agree with this 100%!! In fact I actually proposed this as an issue #83445 before realizing it was already implemented in `scipy.optimize.minimize`. Anyway, since I removed the `newton/secant/halley` part of this PR, I think this point is now moot.